components/component.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef COMPONENT_H_
+#define COMPONENT_H_
+
+#include <string>
+
```

Comment:
Read about namespaces and scoping https://google.github.io/styleguide/cppguide.html#Namespaces